src/util/std_expr.cpp

```diff
@@ -145,6 +146,7 @@ void object_descriptor_exprt::build(
     offset()=from_integer(0, index_type());
 
   build_object_descriptor_rec(ns, expr, *this);
+  simplify(offset(), ns);
```

Comment:
I'm a bit worried about the cost, can you check that callees don't currently run this after invoking build? (And if they do, remove it from those places.)

Reply:
Checked, there was one redundant simplify (presumably working around the same issue). A couple of others use the offset to build a composite expression then simplify *that*, and so would have coincidentally dodged this problem. Some others uses `ode.offset().is_zero()`, which will probably have failed spuriously before when the offset was composite-but-constant.